src/logic/sourcemap.ts

```diff
@@ -0,0 +1,67 @@
+import * as vlq from 'vlq';
+
+export class SourceMap {
+  version: number;
+  sources: string[];
+  names: string[];
+  mappings: string;
+
+  pcToLine: { [key: number]: number };
+  lineToPc: { [key: number]: number[] };
+
+  constructor({
+    version,
+    sources,
+    names,
+    mappings,
+  }: {
+    version: number;
+    sources: string[];
+    names: string[];
+    mappings: string;
+  }) {
+    this.version = version;
+    this.sources = sources;
+    this.names = names;
+    this.mappings = mappings;
+
+    if (this.version !== 3)
+      throw new Error(`Only version 3 is supported, got ${this.version}`);
+
+    if (this.mappings === undefined)
+      throw new Error(
+        'mapping undefined, cannot build source map without `mapping`'
+      );
+
+    const pcList = this.mappings.split(';').map((m) => {
+      const decoded = vlq.decode(m);
+      if (decoded.length > 1) return decoded[2];
```

Comment:
@barnjamin I'm probably missing some intuition, can you double check me?

The highlighted line retrieves the element at index 2.  So, I'd expect the length check to be > 2, but it reads > 1.

Reply:
you're right indeed. 

I think in this case, it happens to be true that if the length > 1 it'll have the 2th element populated but that makes no sense to check. Updated with better check.